onnxruntime/core/graph/graph.cc

```diff
@@ -197,6 +197,14 @@ const TensorShapeProto* NodeArg::Shape() const {
       }
       return nullptr;
     }
+    case TypeProto::kOptionalType: {
+      // Store shape only for optional types that are tensors
+      if (type->optional_type().elem_type().has_tensor_type() &&
```

Comment:
>type->optional_type().elem_type().has_tensor_type() [](http://example.com/codeflow?start=10&length=51)

This either already has utils::Has* method or needs a new one.

Reply:
Added new ones